### ISSUE TITLE
Credentials._setCredentialsFromSession generates invalid provider name

### ISSUE THREAD

**Issue by juranki** [aws-amplify/amplify-js#6462]
`Credentials._setCredentialsFromSession` generates invalid provider name **when identity and user pools are on different regions**

**To Reproduce:**
1. Have Cognito **user and identity pools in different regions**
2. Log in with amplify-js client 
3. Observe the failing `https://cognito-identity.REGION.amazonaws.com/` call
   - Request body has a provider name of form `cognito-idp.eu-west-1.amazonaws.com/eu-central-1_XXXXXXXXX` in `Logins` structure (note the conflicting regions)
   - Response: `NotAuthorizedException: Invalid login token. Issuer doesn't match providerName`

**Expected behavior**

The regions in the provider name should be consistent with the region of the user pool that issued the id token, to successfully produce federated credentials. 

**Code Snippet**

Provider name is constructed here: https://github.com/aws-amplify/amplify-js/blob/aws-amplify%403.0.22/packages/core/src/Credentials.ts#L356

**Screenshots**

![invalid_key](https://user-images.githubusercontent.com/6607/89024997-bfea9680-d32e-11ea-8c24-ec7f0c841d7d.png)

**What is Configured?**

```
"aws-amplify": "^3.0.22",
```

```
  Auth.configure({
    userPoolId: 'eu-central-1_XXXXXXXXX',
    userPoolWebClientId: 'XXXXXXXXXXXXXXXXXXXX',
    identityPoolId: 'eu-west-1:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXXX',
    identityPoolRegion: 'eu-west-1',
    region: 'eu-central-1',
    mandatorySignIn: false,
  });

  Analytics.configure({
    disabled: false,
    AWSPinpoint: {
      appId: 'XXXXXXXXXXXXXXXXXXXXXXXX',
      region: 'eu-west-1',
      mandatorySignIn: false,
    }
  })
```

We don't use Amplify CLI to set up resources; only the javascript client in a browser, so far.

But I think resource configuration is correct, because everything works when I apply these changes: https://github.com/juranki/amplify-js/commit/29af87c566749002c1f3d0a65c801940f8cd1d83

**Comment by ashika01** [comment]
@juranki - this is an old issue on related library - https://github.com/amazon-archives/amazon-cognito-identity-js/issues/50

have you setup everything properly. I think there may be some disconnect.

**Comment by juranki** [comment]
@ashika01 
I think the referred issue is only relevant in the sense that using user and id pools from different regions is possible. amazon-cognito-identity-js is not part of amplify-js and has not been developed in 3 years.

I think my setup is correct, because everything works when I use a patched version of amplify-js. I'm able to get credentials from an id pool in ireland, using a token issued by a user pool in frankfurt. And use those credentials to record pinpoint events in ireland.

On the other hand, when I use an unpatched version of amplify-js, getting credentials from ireland fails. And the failure is caused by an invalid provider name in the GetId -request. amplify-js uses the region of the id pool to construct the provider name of the user pool. And that, clearly, is a **bug**.

ps. I'll change the wording of issue title and body to more closely match the terms used in identity pool documentation

**Comment by ashika01** [comment]
I am not following. What’s the patched version? Are you making some changes to amplify-js and using that?! If so, feel free to open a PR and I can work with you to have this merged that. Also, that library that I linked here, is brought inside amplify-js github now and is maintained.

**Comment by juranki** [comment]
The patched version is one with this change applied: https://github.com/juranki/amplify-js/commit/29af87c566749002c1f3d0a65c801940f8cd1d83

I'm happy to submit that as a pull request if it's selected as the correct way to fix the problem. But it very narrowly focuses on finding the correct provider name in this specific situation. I'm not familiar with amplify-js codebase so I chose a solution that can be applied locally in that one method. 

But why was id pool region used, even if the configuration has entries for the regions of both id and user pools? This line caught my eye: https://github.com/aws-amplify/amplify-js/blob/aws-amplify%403.0.22/packages/auth/src/Auth.ts#L179 

It appears to collapse the id and user pool regions into one setting and might have implications, besides the one I bumped into.

**Comment by ashika01** [comment]
@juranki thank you for the intense research. Okay I got it now. I will look into this and see we can make changes in our end. Thank you 😊

**Comment by juranki** [comment]
@ashika01 Thank you!!

**Comment by ckrodrigues** [comment]
Hey @ashika01 @juranki
I'm having the exact same issue. Do we have a solution for this?
Thanks!

**Comment by cpyle0819** [comment]
@ashika01 also encountering this. Can this be reopened?

**Comment by nhpolanco** [comment]
@ashika01 Cant use amplify until this issue is fixed. Any updates?

**Comment by chrisbonifacio** [comment]
Related to https://github.com/aws-amplify/amplify-js/issues/9409

**Comment by abdallahshaban557** [comment]
Hi @juranki - this issue has now been resolved with version 5.0.5!
